### DISCUSSION TITLE
how edge_index longTensor is related to edge_index with SparseTensor?

### DISCUSSION THREAD

**Discussion by gebrahimi91** [pyg-team/pytorch_geometric#2412] [Q&A] [answered]
In the documentation, it is mentioned:
`        edge_index (Tensor or SparseTensor): A :obj:`torch.LongTensor` or a
            :obj:`torch_sparse.SparseTensor` that defines the underlying graph
            connectivity/message passing flow.
            :obj:`edge_index` holds the indices of a (sparse) symmetric
            adjacency matrix.
            If :obj:`edge_index` is of type :obj:`torch.LongTensor`, its shape
            must be defined as :obj:`[2, num_edges]`, where messages from nodes
            :obj:`edge_index[0]` are sent to nodes in :obj:`edge_index[1]`
            (in case :obj:`flow="source_to_target"`).
            If :obj:`edge_index` is of type :obj:`torch_sparse.SparseTensor`,
            its sparse indices :obj:`(row, col)` should relate to
            :obj:`row = edge_index[1]` and :obj:`col = edge_index[0]`.
            The major difference between both formats is that we need to input
            the *transposed* sparse adjacency matrix.
`
I get how the edge_index works when it uses LongTensor, but not sure how the sparse_version works? Can anyone give me an example of an edge_index Long Tensor and how I can convert it to the SparseTensor version?
For instance, how do you convert this to a SparseTensor?
```

tensor([[ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15],
        [ 0,  0,  0,  3,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0]])
```

**Comment by rusty1s** [comment, marked as answer]
You can read more about it [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/sparse_tensor.html).
